tests/python_compiler/inspection/test_dag_builder.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit test module for the DAGBuilder abstract base class."""
+
+from typing import Any
+
+import pytest
+
```

Comment:
I think we should make the test skip in the absence of `xdsl`.

```suggestion
xdsl = pytest.importorskip("xdsl")
```